tedana/workflows/ica_reclassify.py

```diff
@@ -326,10 +326,11 @@ def ica_reclassify_workflow(
         raise ValueError("The following components were both accepted and rejected: " f"{in_both}")
 
     # boilerplate
-    basename = "report"
+    prefix = prefix + "_" if (prefix != "" and not prefix.endswith("_")) else prefix
+    basename = f"{prefix}report"
```

Comment:
```suggestion
    prefix_full = prefix + "_" if (prefix != "" and not prefix.endswith("_")) else prefix
    basename = f"{prefix_full}report"
```
You're overwriting `prefix` and the original version is later called by the iogenerator. That's why tests are failing.

Reply:
Then this could probably be fixed by changing
https://github.com/tsalo/tedana/blob/10e294e1e6259446cb8feed952509f6ff68bbb1c/tedana/io.py#L134
to
`self.prefix = prefix + "_" if (prefix != "" and not prefix.endswith("_")) else prefix`

Reply:
I'm not sure why, but your new function `_infer_prefix` is outputting `None` instead of `""` for an empty prefix. That means later functions are failing to combine `None+Str` and causing a bunch of failed tests.

Reply:
Thanks for catching that @handwerkerd. The issue is that I forgot to actually return the modified prefix, so it was returning None no matter what.